pkg/image/registry/image/etcd/etcd.go

```diff
@@ -59,8 +59,27 @@ func NewREST(optsGetter restoptions.Getter) (*REST, error) {
 	}
 
 	if err := restoptions.ApplyOptions(optsGetter, store, prefix); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return &REST{store}, nil
+	internalStore := *store
+	internalStore.UpdateStrategy = image.InternalStrategy
+	internalStore.CreateStrategy = nil
+
+	return &REST{store}, &InternalREST{store: &internalStore}, nil
+}
+
+// InternalREST implements a RESTStorage for image resources internally modifiable.
```

Comment:
> I disagree.  If someone can directly mutate the container (images), why shouldn't they be allowed to control the signatures?

As discussed before, we don't want to allow to set signatures during image
push nor image import.

The image rest storage is being used in other virtual rest stores
(ImageStreamMapping, ImageStreamImport). It seems reasonable to me to allow
admin to create/update image directly with any signatures he wants using the
regular store. The other virtual rests should be passed some limited image
rest though.

So instead of adding InternalREST, would it be okay to:

1. add something like UnsignedREST
2. make the image REST fully capable

?